fe/fe-core/src/main/java/org/apache/doris/analysis/Expr.java

```diff
@@ -2375,6 +2375,14 @@ private static boolean customNullableAlgorithm(Function fn, List<Expr> children)
             }
             return false;
         }
+        if (fn.functionName().equalsIgnoreCase("array_reduce")) {
```

Comment:
some rule as  `array_map`, should be || fn.functionName().equalsIgnoreCase("array_reduce")